### DISCUSSION TITLE
Difficulty using Sequel mocking

### DISCUSSION THREAD

**Discussion by mattlindsey** [jeremyevans/sequel#2052] [General]
Quick question, but let me know if you need more info.
My code saves the db in a class variable using `@db = Sequel.connect(connection_string)`, and when testing from rspec I am able create a mock and stub `[]` using this
```ruby
require "sequel"

RSpec.describe Tool::Database do
  subject { described_class.new(connection_string: "mock:///") }
...
      allow(subject.db).to receive(:[])
        .with("SELECT max(salary), count(*) FROM users")
        .and_return(rows)
 ```
 
 That works, but when I try to stub `(:tables)`, I get this error
  
```
     NoMethodError:
       undefined method `tables' for #<Sequel::Mock::Database: "mock:///">
 ```
 
 Am I maybe approaching this wrong?

**Comment by mattlindsey** [comment]
I tried stubbing like this, but not working:
```ruby
       allow_any_instance_of(Sequel::Mock::Database).to receive(:tables)
        .and_return(rows)

**Comment by jeremyevans** [comment]
`Sequel::Mock::Database#tables` is not defined.  Maybe you want to use `mock://postgres` or something similar instead of `mock://`.

Note that I haven't used rspec significantly in about 8 years, and never used the mocking features of rspec much, so I can probably only answer Sequel-specific questions.  For rspec-specific questions, it would be best to ask on an rspec-specific forum.

**Reply by mattlindsey** [reply]
`mock://postgres` works!  Thank you!

**Reply by maxnotarangelo** [reply]
Just chiming in to say this fixed my issue as well. I was using `mock://example`, and running into issues when enabling extensions:

```
> DB.extension(:pg_interval)

NoMethodError: undefined method `add_conversion_proc' for #<Sequel::Mock::Database:0x00007fc9a7253848> (NoMethodError)

          add_conversion_proc(1186, Postgres::IntervalDatabaseMethods::PARSER)
```

But switching to `mock://postgres` fixed it.
